pkgs/development/tools/tabnine/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchurl }:
+
+let
+  hashes = {
+    x86_64-linux   = "caa86a00d0ede8485bf855276ac1b2e411a5e99de612696b137ea3519a8e4967";
+    x86_64-darwin  = "eb5ceb534629dcf5d57640bf672ea0d94ac02e264209ab3a29e6bda8a44d3b8f";
+  };
+in
+stdenv.mkDerivation rec {
+  name = "tabnine-${version}";
+  version = "1.0.12";
+
+  platformString =
+    if stdenv.isDarwin then "x86_64-apple-darwin"
+    else if stdenv.isLinux then "x86_64-unknown-linux-gnu"
+    else throw "unsupported platform";
+
+  src = fetchurl {
+    url = "https://update.tabnine.com/${version}/${platformString}/TabNine";
+    sha256 = hashes.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  };
+
+  phases = "installPhase";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${src} $out/bin/TabNine
+    chmod +x $out/bin/TabNine
+  '';
+
+  meta = with stdenv.lib; {
+    description = "TabNine is the all-language autocompleter. It uses machine learning to provide responsive, reliable, and relevant suggestions.";
+    homepage = https://tabnine.com/;
+    license = stdenv.lib.licenses.unfreeRedistributable;
```

Comment:
How do you know we have permission to redistribute it?

Reply:
That is a good question. I exchanged emails with the author/owner of Tabnine asked him for the URL and sha to include in Nix and told him thats what I am doing and he was very happy I was doing that. Tabnine is free to download though proprietary binary, you get a paid license to unlock pro features. Hence I thought above license was a fit and the fact author helped me  with information I needed and was happy I was adding it the above license was the best fit. If there is a license that is a better fit let me know. 

Another option is I add his EULA as a new license to nix: 

https://tabnine.com/eula

What are your thoughts? What do you prefer me to do with this?

Reply:
Using `unfreeRedistributable` sounds good then. Can you add a small quote from the TabNine author regarding that as a comment above the license?